src/plugins/FullScreenPlugin.js

```diff
@@ -0,0 +1,114 @@
+import { ButtonPlugin } from '../base-plugins/ButtonPlugin';
+import { Button } from '../ui-elements';
+
+/**
+ * A Springroll plugin to easily set up togglable fullscreen
+ */
+export class FullScreenPlugin extends ButtonPlugin {
+  
+  /**
+   *  Creates an instance of FullscreenPlugin
+   * 
+   * @param {string} targetElementSelector The selector for the element to be made fullscreen
+   * @param {string || HTMLElement } buttonSelector The selector for the button or the HTMLElement Button to be used to toggle fullscreen on the targetElement
+   */
+  constructor(buttonSelector) {
+    super({
+      name: FullScreenPlugin.fullscreenKey
+    }); 
+
+    this._toggleButtons = [];
+    this.iFrame = null;
+
+    this.sendAllProperties = this.sendAllProperties.bind(this);
+
+    
+    this.toggleButton = document.querySelectorAll(buttonSelector);
+    this.toggleButton.forEach((button) => {
+      this._toggleButtons.push(new Button({
+        button: button,
+        onClick: this.toggleFullScreen.bind(this),
+        channel: FullScreenPlugin.fullscreenKey
+      }));
+    });
+    
+
+    document.addEventListener('fullscreenchange',  () => {
+      this.sendAllProperties(FullScreenPlugin.fullscreenKey, this.isFullScreen);
```

Comment:
sendAllProperties doesn't actually need any input its implemented down on line 75, it just sends the currently active property.

Reply:
But the sendAllProperties call doesn't actually use the `isFullScreen` get so it might be worth updating that just to keep everything consistent.